Environment variable support for mi components (#767)
wso2/docs-choreo-dev: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md

## Environment variables

> It is recommended to use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.

Reviewer: Are you trying to say that we recommend that they use environment variables? or is it as an alternative to another option?

Author: We recommend that they use environment. It is the better option here.

> The component deploy page offers a user-friendly interface for managing environment variables across different environments. For more advanced use cases, the Choreo DevOps portal is available.

Reviewer: Would you be able to explain what message we need to convey here?

Author: The users can manage environment variables for the general use cases through the component deployment page. If they have more advanced use cases, they can do the configurations in Choreo DevOps portal.

Reviewer: The component's **Deploy** page allows you to easily manage environment variables across environments via a user-friendly interface. For more information on how you can manage environment variables for more advanced use cases, see [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets).

Author: Fixed in a1a9d6250d9d356ba65d8826e0ee55a40d3a258e

## Environment variables

> You can use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.

Reviewer 2: ```suggestion
It is recommended to use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.
```

Author: fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

> The Choreo Console allows you to define environment variables for your integration project. You can define the environment variables before you deploy the component.

Reviewer 2: ```suggestion
The component deploy page offers a user-friendly interface for managing environment variables across different environments. For more advanced use cases, the Choreo DevOps portal is available.

The Choreo DevOps Portal allows for greater flexibility in defining configurations and secrets for deployed integrations, as well as other component types, to meet more complex configuration requirements. Refer to the section on Configurations and secrets for further details.
```

Author: Fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

![Environment Variables](../../assets/img/ipaas/integration-studio/env_variables.png){.cInlineImage-half}

> The Choreo DevOps Portal also allows you to define configurations and secrets to a deployed integration (as well as other component types) with more flexibility to cater your advance configuration needs. For more information, see [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets).

Reviewer 2: ```suggestion
The Choreo DevOps Portal allows for greater flexibility in defining configurations and secrets for deployed integrations, as well as other component types, to meet more complex configuration requirements. Refer to the section on [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets) for further details.
```

Author: Fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

> For a list of parameters that you can configure as environment variables, see [WSO2 API Manager Documentation - Injecting Parameters - Supported parameters](https://apim.docs.wso2.com/en/latest/integrate/develop/injecting-parameters/#supported-parameters).

Reviewer 2: ```suggestion
For a comprehensive list of parameters that can be configured as environment variables, please refer to the [WSO2 API Manager Documentation - Injecting Parameters - Supported parameters](https://apim.docs.wso2.com/en/latest/integrate/develop/injecting-parameters/#supported-parameters).
```

Author: Fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

> You can use environment variables in WSO2 Micro Integrator to improve the configuration management, security, portability, and manageability of the Micro Integrator installation. By using environment variables, organizations can simplify the management and maintenance of their micro installations and ensure that they can be quickly and easily updated when required.
> 

Reviewer 2: ```suggestion
You can use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.

```

> The Choreo Console allows you to define enviornment variables for your integration project. You can define the environment variables before you deploy the component.

Reviewer 2: ```suggestion
The Choreo Console allows you to define environment variables for your integration project. You can define the environment variables before you deploy the component.
```

Author: Fixed in fd88d99a57c17c5241a87960c3e13d754445543a
